dissect/target/plugins/os/unix/log/auth.py

```diff
@@ -1,39 +1,264 @@
 import re
+from datetime import datetime
+from enum import Enum
+from functools import lru_cache
 from itertools import chain
-from typing import Iterator
+from pathlib import Path
+from typing import Iterator, Union
 
+from dissect.target import Target
 from dissect.target.exceptions import UnsupportedPluginError
-from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.helpers.record import DynamicDescriptor, TargetRecordDescriptor
 from dissect.target.helpers.utils import year_rollover_helper
 from dissect.target.plugin import Plugin, export
 
-AuthLogRecord = TargetRecordDescriptor(
-    "linux/log/auth",
-    [
-        ("datetime", "ts"),
-        ("string", "message"),
-        ("path", "source"),
-    ],
-)
-
 _TS_REGEX = r"^[A-Za-z]{3}\s*[0-9]{1,2}\s[0-9]{1,2}:[0-9]{2}:[0-9]{2}"
 RE_TS = re.compile(_TS_REGEX)
 RE_TS_AND_HOSTNAME = re.compile(_TS_REGEX + r"\s\S+\s")
 
 
+class AuthLogServicesEnum(str, Enum):
+    cron = "CRON"
+    su = "su"
+    sudo = "sudo"
+    sshd = "sshd"
+    systemd = "systemd"
+    systemd_logind = "systemd-logind"
+    pkexec = "pkexec"
+
+
+class AuthLogRecordBuilder:
```

Comment:
I'm not really a fan of this "god class". I'd prefer smaller individual functions or classes for each "enhanced" parser.